crates/solidity/outputs/cargo/tests/src/bindings_assertions/assertions.rs

```diff
@@ -415,28 +431,24 @@ fn check_reference_assertion(
     Ok(())
 }
 
-fn find_and_resolve_reference(
-    bindings: &Bindings,
-    assertion: &ReferenceAssertion,
-) -> Result<Option<Cursor>, String> {
+fn find_and_resolve_reference<'a>(
+    bindings: &'a Bindings,
+    assertion: &ReferenceAssertion<'_>,
+) -> Result<Option<Definition<'a>>, String> {
     let ReferenceAssertion { cursor, .. } = assertion;
 
-    let Some(handle) = bindings.cursor_to_handle(cursor) else {
+    let Some(reference) = bindings.reference_at(cursor) else {
         return Err(format!("{assertion} failed: not found"));
     };
-    if !handle.is_reference() {
-        return Err(format!("{assertion} failed: not a reference"));
-    }
-    Ok(handle
-        .jump_to_definition()
-        .map(|handle| handle.get_cursor().unwrap()))
+
+    Ok(reference.jump_to_definition())
 }
 
-fn lookup_referenced_definition(
-    bindings: &Bindings,
-    definitions: &HashMap<String, DefinitionAssertion>,
-    assertion: &ReferenceAssertion,
-) -> Result<Cursor, String> {
+fn lookup_referenced_definition<'a>(
```

Comment:
Not suggesting we make any changes in this PR, but just asking for my own understanding:

the `bindings_assertions` system is growing quickly over time, and increasing in complexity, compared to the actual source it is testing..
I wonder if we still think it is better for testing certain scenarios rather than just adding `bindings_output` tests and checking everything at once. Is the maintenance effort worth it?

Reply:
Generally speaking, anything that automates robustness checking is worth it IMHO. Snapshots are fine, but require manual inspection on each PR. Assertions have already pointed me to a regression I was inadvertently introducing when removing the ellipsis operator.

If anything, I would probably do the opposite and remove the snapshots, probably generating Mermaid for the assertion samples for debugging. Snapshots have been useful to somehow measure progress by seeing `unresolved` labels turn into actual definitions, but so far, at least for me, they are less useful.